### PR TITLE
Scheduler - Make Form as a base for RecurrenceEditor

### PR DESCRIPTION
**Pull request by Yarovaya**

### CODE REVIEW

File: js/ui/scheduler/ui.scheduler.recurrence_editor.js [hunk 1]
```diff
@@ -199,373 +195,311 @@ const RecurrenceEditor = Editor.inherit({
...
+        for(let i = 0; i < 12; i++) {
+            months[i] = { value: String(i + 1), text: monthsNames[i] };
+        }
 
-        const interval = this._recurrenceRule.rules().interval || 1;
+        return [
+            {
+                dataField: 'byday',
+                colSpan: 2,
+                template: (_, itemElement) =>{
```
Comment (AntonSermyazhko) [inline js/ui/scheduler/ui.scheduler.recurrence_editor.js]: To my mind `Template` should be in a separate method.

File: js/ui/scheduler/ui.scheduler.recurrence_editor.js [hunk 2]
```diff
@@ -199,373 +195,311 @@ const RecurrenceEditor = Editor.inherit({
...
+                },
+                visible: freq === 'monthly' || freq === 'yearly',
+                label: {
+                    visible: false
+                }
+            }
+        ];
+    },
+
+    _getRepeatEndItems: function() {
```
Comment (AntonSermyazhko) [inline js/ui/scheduler/ui.scheduler.recurrence_editor.js]: Is it possible to create this items without template by using form items?
@jsdmitry what do you think?

File: js/ui/scheduler/ui.scheduler.recurrence_editor.js [hunk 3]
```diff
@@ -199,373 +195,311 @@ const RecurrenceEditor = Editor.inherit({
...
-            .addClass(INTERVAL_EDITOR)
-            .addClass(FIELD_VALUE_CLASS);
-
-        const $intervalEditorLabel = $('<div>')
-            .text(messageLocalization.format('dxScheduler-recurrenceRepeatEvery'))
-            .addClass(INTERVAL_EDITOR + LABEL_POSTFIX)
-            .addClass(FIELD_LABEL_CLASS);
+    _getRepeatOnItems: function(freq) {
+        const monthsNames = dateLocalization.getMonthNames('wide');
+        const months = [];
```
Comment (jsdmitry) [inline js/ui/scheduler/ui.scheduler.recurrence_editor.js]: Can simplify
`const months = [...Array(12)].map((_, i) => ({ value: `${i + 1}`, text: monthNames[i] }))`

File: js/ui/scheduler/ui.scheduler.recurrence_editor.js (hunk 2 above)
Comment (Yarovaya) [reply js/ui/scheduler/ui.scheduler.recurrence_editor.js]: RadioGroup is a predefined type of editor for Form and thus I can set itemTemplate in editor options. I will correct it!

File: js/ui/scheduler/ui.scheduler.recurrence_editor.js (hunk 1 above)
Comment (Yarovaya) [reply js/ui/scheduler/ui.scheduler.recurrence_editor.js]: Yes, it would be wise to fix it

File: styles/widgets/common/scheduler.less [hunk 4]
```diff
@@ -25,6 +25,12 @@
     .dx-scrollable-content {
         padding-top: @SCHEDULER_POPUP_SCROLLABLE_CONTENT_PADDING;
     }
+
+    .dx-form.dx-recurrence-editor-container {
```
Comment (AntonSermyazhko) [inline styles/widgets/common/scheduler.less:29]: Can we collect the all recurrence editor CSS (except theme-based CSS) in the `recurrenceEditor.less` file?

**Review by AntonSermyazhko — DISMISSED** [review]

File: styles/widgets/common/scheduler.less (hunk 4 above)
Comment (Yarovaya) [reply styles/widgets/common/scheduler.less:29]: Yes, we already have such files. And for theme-based css too! But this code is in .dx-scheduler-appointment-popup parent section.

File: js/ui/scheduler/ui.scheduler.recurrence_editor.js (hunk 3 above)
Comment (AntonSermyazhko) [reply js/ui/scheduler/ui.scheduler.recurrence_editor.js]: Sorry, but It is not "simplify" ))
How to read or debug this piece of code?

**Review by AntonSermyazhko — APPROVED** [review]

**Review by AryamnovEugeniy — APPROVED** [review]

**Review by andrewmakarov — APPROVED** [review]

**Review by jsdmitry — APPROVED** [review]

**Pull request merged**